### PR TITLE
[Playwright] speculos // swap e2e tests with common currencies (BTC, USDT, ETH)

### PR DESCRIPTION
**Pull request by bharamboure-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [x] **Covered by automatic tests.** (actually automated tests)
- [ ] **Impact of the changes:** none, just more swap tests scenarios in allure reporting

### 📝 Description

Swap test cases between most used currencies (BTC, SOL, ETH, USDT)
More to come

Jira test ids for XRAY mapping are missing intentionally (don't have those tickets yet)

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/QAA-274
- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/QAA-279
- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/QAA-280

[allure reporting](https://ledger-live.allure.green.ledgerlabs.net/allure/reports/1e8451a5-015a-41f4-b52e-9215a7e9ee7f/)

https://github.com/user-attachments/assets/e345cc6b-11f4-493f-9323-598133bc3b36

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts [hunk 1]
```diff
@@ -4,36 +4,94 @@ import { AppInfos } from "tests/enum/AppInfos";
 import { setExchangeDependencies } from "@ledgerhq/live-common/e2e/speculos";
 import { Fee } from "tests/enum/Fee";
 import { Swap } from "tests/models/Swap";
-import { Providers, Rates } from "tests/enum/Swap";
+import { Provider, Rates } from "tests/enum/Swap";
 
 const swaps = [
   {
-    swap: new Swap(Account.ETH_1, Account.BTC_NATIVE_SEGWIT_1, "0.015", Fee.MEDIUM),
+    swap: new Swap(
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:11]: Please attach corresponding test cases to every test in this file

Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:11]: still waiting for those :sadge:
I'll create them this afternoon
cc @gbecerra-ledger @stual-ledger

File: apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts [hunk 2]
```diff
@@ -4,36 +4,94 @@ import { AppInfos } from "tests/enum/AppInfos";
...
+      Account.BTC_NATIVE_SEGWIT_1,
+      Account.ETH_1,
+      "0.00067",
+      Fee.MEDIUM,
+      Provider.CHANGELLY,
+    ),
+  },
+  {
+    swap: new Swap(Account.ETH_USDT_1, Account.ETH_1, "35", Fee.MEDIUM, Provider.CHANGELLY),
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:33]: this case is failing on my side on quote exchange selection step

File: apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts (hunk 1 above)
Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:13]: done just now

File: apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts [hunk 3]
```diff
@@ -4,65 +4,139 @@ import { AppInfos } from "tests/enum/AppInfos";
...
+      Account.BTC_NATIVE_SEGWIT_1,
+      Account.SOL_1,
+      "0.0006",
+      Fee.MEDIUM,
+      Provider.CHANGELLY,
+    ),
+    xrayTicket: "B2CQA-2747",
+  },
+  {
+    swap: new Swap(
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:55]: this case is failing on my side with insufficient funds error

File: apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts [hunk 4]
```diff
@@ -4,65 +4,139 @@ import { AppInfos } from "tests/enum/AppInfos";
...
+      Account.BTC_NATIVE_SEGWIT_1,
+      Account.ETH_USDT_1,
+      "0.0006",
+      Fee.MEDIUM,
+      Provider.CHANGELLY,
+    ),
+    xrayTicket: "B2CQA-2746",
+  },
+  {
+    swap: new Swap(
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/swap.spec.ts:65]: this case is failing on my side on quote exchange selection step

File: apps/ledger-live-desktop/tests/page/swap.page.ts [hunk 5]
```diff
@@ -115,10 +116,7 @@ export class SwapPage extends AppPage {
   }
 
   @step("Select exchange quote $0 with rate $1")
-  async selectExchangeQuote(
-    providerName: "changelly" | "cic" | "oneinch" | "paraswap",
```
Comment (mbertin-ledger) [inline apps/ledger-live-desktop/tests/page/swap.page.ts:119]: [Comment] I don't know if it's normal but it you look at the providerinside `Swap.ts` file there is only 2 provider (changelly and oneinch).  So `CIC`and `Paraswap` providers are missing.

Comment (bharamboure-ledger) [reply apps/ledger-live-desktop/tests/page/swap.page.ts:119]: atm those are the only ones I use in my tests scenarios, there are more to come in another PR with `EXODUS` for instance
nonetheless, previously you were limited to 4 values, which isn't the case anymore
also won't use providers that require registration or use their own webview for the time being

**Review by hedi-edelbloute — APPROVED** [review]

**Review by mbertin-ledger — APPROVED** [review]

**Review by ypolishchuk-ledger — DISMISSED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

**Review by mbertin-ledger — APPROVED** [review]

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -8,15 +8,15 @@ Please make sure to read CONTRIBUTING.md if you have not already. Pull Requests
...
+- [ ] **Covered by automatic tests.** <!-- if not, please explain. (Feature must be tested / Bug fix must bring non-regression) -->
 - [ ] **Impact of the changes:** <!-- Please take some time to list the impact & what specific areas Quality Assurance (QA) should focus on -->
   - ...
 
 ### 📝 Description
 
 _Replace this text by a clear and concise description of what this pull request is about and why it is needed. Be sure to explain the problem you're addressing and the solution you're proposing._
 _For libraries, you can add a code sample of how to use it._
-_For bugfixes, you can explain the previous behavior and how it was fixed._
+_For bug fixes, you can explain the previous behaviour and how it was fixed._
```
Comment (themooneer) [inline .github/pull_request_template.md:19]: as i assume us locale uses behavior

**Review by themooneer — APPROVED** [review]

**Review by ypolishchuk-ledger — APPROVED** [review]

File: .github/pull_request_template.md (hunk 6 above)
Comment (bharamboure-ledger) [reply .github/pull_request_template.md:19]: will edit / approve if any complain 😄

**Pull request merged**